### PR TITLE
Allow for probe timeouts to be configurable

### LINKED ISSUE #11109: Allow For Probe Timeouts to Be Configurable
### What
Currently, the Teleport cluster helm chart does not allow for configurable timeouts for `readinessProbe.timeoutSeconds` and `livenessProbe.timeoutSeconds`.

### How
I would update the `teleport-cluster` helm chart to include a property that allows for this to be configured.

### Why
I've run into [this issue](https://github.com/kubernetes/kubernetes/issues/89898) when creating a new GCP Kubernetes cluster. 

### Workaround
Forking the helm chart and making these changes.

### PR DESCRIPTION
**Pull request by hmadison**
When setting up a new Teleport enterprise cluster on GCP,
I noticed that I needed to set the probe timeouts to get the
cluster to be helthy. This seems to be a known issue (https://github.com/kubernetes/kubernetes/issues/89898).

As a "stopgap", I've updated the helm chart to allow for end users
to be able to configure these timeouts.

Closes #11109

### CODE REVIEW

File: examples/chart/teleport-cluster/values.yaml [hunk 1]
```diff
@@ -263,3 +263,7 @@ priorityClassName: ""
 # Tolerations for pod assignment
 # https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
 tolerations: []
+
+# Timeouts for the readiness and liveness probes
+# https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+probeTimeout: 1
```
Comment (codingllama) [inline examples/chart/teleport-cluster/values.yaml:269]: Suggestion: mention time unit, bump defaults a bit.

```suggestion
probeTimeoutSeconds: 5
```

**Review by codingllama — COMMENTED** [review]
Thanks for the contribution, @hmadison!

**Comment by codingllama** [comment]
@zmb3 and @r0mant, should we look for parity in other charts after this one lands? Seems like a sensible setting to have.

**Comment by zmb3** [comment]
> @zmb3 and @r0mant, should we look for parity in other charts after this one lands? Seems like a sensible setting to have.

Seems reasonable to me. @webvictim - thoughts?

**Review by webvictim — CHANGES_REQUESTED** [review]
Agree with @codingllama's comment regarding making the value a little easier to understand by using a unit in the name.

@hmadison There's also a few other things we'll need before we can approve/merge/backport (sorry, haven't got around to writing the contribution guidelines yet - they're in the works):

1) Please add `probeTimeoutSeconds` at the appropriate location in the `teleport-cluster` chart reference: https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx

2) Please add `probeTimeoutSeconds` to https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/values.schema.json

3) Please add an example which sets `probeTimeoutSeconds` so it can be validated by the linter - see other examples in https://github.com/gravitational/teleport/tree/master/examples/chart/teleport-cluster/.lint

Thank you!

**Comment by hmadison** [comment]
@webvictim / @codingllama I've made the requested changes.

**Review by codingllama — APPROVED** [review]
LGTM. Thanks again for contributing!

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 2]
```diff
@@ -2614,3 +2614,26 @@ Kubernetes Tolerations to set for pod assignment.
   ```
   </TabItem>
 </Tabs>
+
+## `probeTimeoutSeconds`
+
+| Type | Default value | Can be used in `custom` mode? |
+| - | - | - |
+| `integer` | `1` | ✅  |
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference.mdx:2622]: Teeny tiny space nit

```suggestion
| `integer` | `1` | ✅ |
```

**Review by webvictim — APPROVED** [review]
LGTM - thanks!

**Review by russjones — APPROVED** [review]

**Comment by webvictim** [comment]
@hmadison For ease, I've opened my own PR (#11396) which implements this feature for both Helm charts, adds unit tests and rebases against the new reference updates we've done. I'll push this through and get it backported to `branch/v9` so it comes out in the next Helm chart release.

Thanks for your contribution!

**Pull request closed without merging**